src/app/courses/progress-courses/courses-progress-leader.component.ts

```diff
@@ -38,7 +38,13 @@ export class CoursesProgressLeaderComponent implements OnInit, OnDestroy {
       const fillEmptyAnswers = (answers) => [].concat(answers, Array(questionsLength - answers.length).fill(''));
       this.submissions = submissions.map(
         submission => {
-          const answers = fillEmptyAnswers(submission.answers.map(a => ({ ...a, mistakes: a.mistakes || 1 - a.grade })));
+          const answers = fillEmptyAnswers(submission.answers.map(a => {
+            const data = { ...a, mistakes: a.mistakes || 1 - a.grade };
```

Comment:
`mistakes` and `grade` should be a number or undefined, so simply adding ` || 0` to the original code should produce the same effect.

Reply:
nope, not working, even my branch
![image](https://user-images.githubusercontent.com/2650479/41801354-7ef70d1c-7650-11e8-91a8-2ce18bbad12c.png)


so I need to find another workaround

Reply:
:) just figured it out.. lol.. @paulbert fixed :)